lib/iris/plot.py

```diff
@@ -344,16 +336,17 @@ def _draw_2d_from_bounds(draw_method_name, cube, *args, **kwargs):
     two_dim_contig_atol = kwargs.pop('two_dim_coord_contiguity_atol',
                                      1e-4)
     for coord in plot_defn.coords:
-        if coord.ndim == 2 and coord.has_bounds():
```

Comment:
Check that `coord` is a coordinate first as coord could be either `None` or an integer

Reply:
But now you're checking twice whether the coord has bounds or not.  One of the checks can be removed.